python_modules/libraries/dagster-components/dagster_components/core/component.py

```diff
@@ -161,7 +165,7 @@ def load_component_type(object_key: LibraryObjectKey) -> type[Component]:
         raise DagsterError(f"Error loading module `{module_name}`.") from e
 
 
-def discover_entry_point_library_objects() -> dict[LibraryObjectKey, type[Component]]:
+def discover_entry_point_library_objects() -> dict[LibraryObjectKey, object]:
```

Comment:
I think we should have a `Union` type alias `LibraryObject` that explicitly lists viable library object types.